bitcoin/examples/taproot-psbt-simple.rs

```diff
@@ -0,0 +1,249 @@
+    let xpriv_input_2 = get_internal_address_xpriv(&secp, master_xpriv, 0);
+    let xpriv_change = get_internal_address_xpriv(&secp, master_xpriv, 1);
+
+    // Get the PKs
+    let (pk_input_1, _) = Xpub::from_priv(&secp, &xpriv_input_1).public_key.x_only_public_key();
+    let (pk_input_2, _) = Xpub::from_priv(&secp, &xpriv_input_2).public_key.x_only_public_key();
+    let (pk_change, _) = Xpub::from_priv(&secp, &xpriv_change).public_key.x_only_public_key();
+
+    // Get the Tap Key Origins
+    // Map of tap root X-only keys to origin info and leaf hashes contained in it.
+    let origin_input_1 = get_tap_key_origin(
+        pk_input_1,
+        Fingerprint::from_str(MASTER_FINGERPRINT).unwrap(),
+        DerivationPath::from_str("m/86'/0'/0'/0/0").unwrap(),
+    );
+    let origin_input_2 = get_tap_key_origin(
+        pk_input_2,
+        Fingerprint::from_str(MASTER_FINGERPRINT).unwrap(),
+        DerivationPath::from_str("m/86'/0'/0'/1/0").unwrap(),
+    );
+    let origins = [origin_input_1, origin_input_2];
+
+    // Get the unspent outputs that is locked to the key above that we control.
+    // In a real application these would come from the chain.
+    let utxos: Vec<TxOut> =
+        dummy_unspent_transaction_outputs().into_iter().map(|(_, utxo)| utxo).collect();
+
+    // Get the addresses to send to.
+    let address = receivers_address();
+
+    // The inputs for the transaction we are constructing.
+    let inputs: Vec<TxIn> = dummy_unspent_transaction_outputs()
+        .into_iter()
+        .map(|(outpoint, _)| TxIn {
+            previous_output: outpoint,
+            script_sig: ScriptBuf::default(),
+            sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+            witness: Witness::default(),
+        })
+        .collect();
+
+    // The spend output is locked to a key controlled by the receiver.
+    let spend = TxOut { value: SPEND_AMOUNT, script_pubkey: address.script_pubkey() };
+
+    // The change output is locked to a key controlled by us.
+    let change = TxOut {
+        value: CHANGE_AMOUNT,
+        script_pubkey: ScriptBuf::new_p2tr(&secp, pk_change, None), // Change comes back to us.
+    };
+
+    // The transaction we want to sign and broadcast.
+    let unsigned_tx = Transaction {
+        version: transaction::Version::TWO,  // Post BIP 68.
+        lock_time: absolute::LockTime::ZERO, // Ignore the locktime.
+        input: inputs,                       // Input is 0-indexed.
+        output: vec![spend, change],         // Outputs, order does not matter.
+    };
+
+    // Now we'll start the PSBT workflow.
+    // Step 1: Creator role; that creates,
+    // and add inputs and outputs to the PSBT.
+    let mut psbt = Psbt::from_unsigned_tx(unsigned_tx.clone()).expect("Could not create PSBT");
+
+    // Step 2:Updater role; that adds additional
+    // information to the PSBT.
+    let ty = PsbtSighashType::from_str("SIGHASH_ALL").expect("valid sighash type");
```

Comment:
Lol, we should definitely have associated constants on `PsbtSighashType`.

Reply:
#2751